applications/plugins/SofaPython/Binding_Data.cpp

```diff
@@ -670,6 +674,26 @@ bool SetDataValuePython(BaseData* data, PyObject* args)
 }
 
 
+SP_CLASS_ATTR_GET(Data, fullPath)(PyObject *self, void*)
+{
+    BaseData* data = ((PyPtr<BaseData>*)self)->object; // TODO: check dynamic cast
+
+    if (sofa::core::objectmodel::DataFileName* dataFilename = dynamic_cast<sofa::core::objectmodel::DataFileName*>(data))
+    {
```

Comment:
Rather than perform a dynamic_cast on DataFileName in generic binding_Data.
The clean way is to create a new binding for DataFileName and to add it to the Factory.